go/tasks/plugins/k8s/spark/spark_test.go

```diff
@@ -353,7 +353,47 @@ func TestBuildResourceSpark(t *testing.T) {
 	dnsOptVal1 := "1"
 	dnsOptVal2 := "1"
 	dnsOptVal3 := "3"
+
+	// Set scheduler
+	schedulerName := "custom-scheduler"
+
+	// Node selectors
+	defaultNodeSelector := map[string]string{
+		"x/default": "true",
+	}
+	interruptibleNodeSelector := map[string]string{
+		"x/interruptible": "true",
+	}
+
+	defaultPodHostNetwork := true
+
+	defaultEnvVars := make(map[string]string)
+	defaultEnvVars["foo"] = "bar"
```

Comment:
@hamersaw do you know how `defaultEnvVars` are carried over? I added the test but did not carry them over.

Reply:
tested locally and yes - the `DecorateEnvVars` bring over the defaultEnvVars.